encord/objects/ontology_labels_impl.py

```diff
@@ -801,21 +801,58 @@ def add_classification_instance(self, classification_instance: ClassificationIns
             )
 
         classification_hash = classification_instance.classification_hash
-        frames = set(_frame_views_to_frame_numbers(classification_instance.get_annotations()))
-        already_present_frames = self._is_classification_already_present(
-            classification_instance.ontology_item,
-            frames,
-        )
         if classification_hash in self._classifications_map and not force:
             raise LabelRowError(
                 f"A ClassificationInstance for classification hash '{classification_hash}' already exists on the label row object. "
                 f"Pass 'force=True' to override it."
             )
 
-        if already_present_frames and not force:
+        """
+        Implementation here diverges because audio data will operate on ranges, whereas
+        everything else will operate on frames.
+        """
+        if self.data_type == DataType.AUDIO:
+            self._add_classification_instance_for_range(
+                classification_instance=classification_instance,
+                force=force,
+            )
+        else:
+            frames = set(_frame_views_to_frame_numbers(classification_instance.get_annotations()))
```

Comment:
Consider refactoring this whole branch into a fn: _add_classification_instance_for_frames